pontoon/base/migrations/0040_remove_changed_entity_locale_entries_for_repository_projects.py

```diff
@@ -0,0 +1,25 @@
+# Generated by Django 3.2.15 on 2023-02-23 16:02
+from django.db import migrations
+from pontoon.base.models import Project, Resource, Entity, ChangedEntityLocale
+
+
+def remove_changed_entity_locale_entries_for_repository_projects(apps, schema_editor):
+    projects = Project.objects.filter(data_source=Project.DataSource.DATABASE)
+    resources = Resource.objects.filter(project__in=projects)
+    entities = Entity.objects.filter(resource__in=resources)
+    ChangedEntityLocale.objects.filter(entity__in=entities).delete()
```

Comment:
You can [follow relationship](https://docs.djangoproject.com/en/4.1/topics/db/queries/#lookups-that-span-relationships) in QuerySet lookups:

```python
ChangedEntityLocale.objects.filter(
    entity__resource__project__data_source=Project.DataSource.DATABASE
).delete()
```